src/robot/utils/misc.py

```diff
@@ -115,3 +115,20 @@ def isatty(stream):
         return stream.isatty()
     except ValueError:    # Occurs if file is closed.
         return False
+
+def parse_re_flags(flags_as_str):
+    result = 0
+    if flags_as_str:
+        unknown = []
+        for f in flags_as_str.split('|'):
+            try:
+                flag = getattr(re, f.upper().strip())
+                if isinstance(flag, re.RegexFlag):
+                    result |= flag
+                else:
+                    unknown.append(f)
+            except AttributeError:
+                unknown.append(f)
+        if unknown:
+            raise ValueError('unknown flag{}: {}'.format(plural_or_not(unknown), seq2str(unknown)))
```

Comment:
- Should use f-string here.
- Error messages should be proper English and thus should start with `U` not `u`.
- Could consider changing the message to 'Unknown regexp flag' to make it more explicit.
- It's nice to report all unknown tags in one go, but I'm not sure how much benefits it actually has over just raising this exception when the first unknown value is encountered. Since you already have implemented it, we can as well keep it.